src/invite-api.js

```diff
@@ -1,236 +1,440 @@
+
+    const pendingInvite = this.#byInviteId.get(inviteIdString)
+    if (!pendingInvite) return false
+
+    this.#byInviteId.delete(inviteIdString)
+
+    const {
+      invite: { projectPublicId },
+    } = pendingInvite
+    const newPendingInvitesForThisProject = (
+      this.#byProjectPublicId.get(projectPublicId) ?? []
+    ).filter((pendingInvite) => pendingInvite.invite.inviteId.equals(inviteId))
+    if (newPendingInvitesForThisProject.length) {
+      this.#byProjectPublicId.set(
+        projectPublicId,
+        newPendingInvitesForThisProject
+      )
+    } else {
+      this.#byProjectPublicId.delete(projectPublicId)
+    }
+
+    return true
+  }
+
+  /**
+   * @param {string} projectPublicId
+   * @returns {PendingInvite[]} the pending invites that were deleted
+   */
+  deleteByProjectPublicId(projectPublicId) {
+    const pendingInvitesDeleted =
+      this.#byProjectPublicId.get(projectPublicId) ?? []
+
+    for (const { invite } of pendingInvitesDeleted) {
+      const inviteIdString = keyToId(invite.inviteId)
+      this.#byInviteId.delete(inviteIdString)
+    }
+
+    this.#byProjectPublicId.delete(projectPublicId)
+
+    return pendingInvitesDeleted
   }
 }
 
+/**
+ * @typedef {Object} InviteApiEvents
+ * @property {(invite: Invite) => void} invite-received
+ * @property {(invite: Invite) => void} invite-removed
+ */
+
 /**
  * @extends {TypedEmitter<InviteApiEvents>}
  */
 export class InviteApi extends TypedEmitter {
-  // Maps project public id -> project id
-  /** @type {Map<string, string>} */
-  #projectIdsMapping = new Map()
-
-  // Maps project id -> set of device ids
-  /** @type {MapOfSets<string, string>} */
-  #peersToRespondTo = new MapOfSets()
-
-  // Maps project id -> { invite, fromPeerId }
-  /** @type {Map<string, { fromPeerId: string, invite: import('./generated/rpc.js').Invite }>} */
-  #pendingInvites = new Map()
-
   #isMember
   #addProject
+  #pendingInvites
 
   /**
    * @param {Object} options
    * @param {import('./local-peers.js').LocalPeers} options.rpc
    * @param {object} options.queries
-   * @param {(projectId: string) => Promise<boolean>} options.queries.isMember
-   * @param {(invite: import('./generated/rpc.js').Invite) => Promise<void>} options.queries.addProject
+   * @param {(projectId: string) => boolean} options.queries.isMember
+   * @param {(projectDetails: Pick<ProjectJoinDetails, 'projectKey' | 'encryptionKeys' | 'projectName'>) => Promise<unknown>} options.queries.addProject
    */
   constructor({ rpc, queries }) {
     super()
     this.rpc = rpc
     this.#isMember = queries.isMember
     this.#addProject = queries.addProject
 
-    this.rpc.on('invite', (peerId, invite) => {
-      this.#handleInvite(peerId, invite).catch(() => {
-        /* c8 ignore next */
-        // TODO: Log errors, but otherwise ignore them, but can't think of a reason there would be an error here
-      })
-    })
+    this.rpc.on('invite', this.#handleInvite)
```

Comment:
I think thrown errors in #handleInvite are no longer being caught, but they should be, to avoid uncaught errors.